sdk/state/integrationtests/state_test.go

```diff
@@ -42,37 +42,46 @@ func TestOpenUpdatesUncoordinatedClose(t *testing.T) {
 
 	// Open
 	t.Log("Open...")
+	// I signs txClose
 	open, err := initiatorChannel.ProposeOpen(state.OpenParams{Asset: asset, AssetLimit: assetLimit})
 	require.NoError(t, err)
-	for {
+	{
 		var authorizedR bool
+		// R signs txClose and txDecl
 		open, authorizedR, err = responderChannel.ConfirmOpen(open)
-		if err != nil {
-			t.Fatal(err)
-		}
+		require.NoError(t, err)
+		require.False(t, authorizedR)
+
 		var authorizedI bool
+		// I signs txDecl and F
 		open, authorizedI, err = initiatorChannel.ConfirmOpen(open)
-		if err != nil {
-			t.Fatal(err)
-		}
-		if authorizedI && authorizedR {
-			break
-		}
+		require.NoError(t, err)
+		require.False(t, authorizedI)
+
+		// R signs F, R is done
+		open, authorizedR, err = responderChannel.ConfirmOpen(open)
+		require.NoError(t, err)
+		require.True(t, authorizedR)
+
+		// I receives the last signatures for F, I is done
+		_, authorizedI, err = initiatorChannel.ConfirmOpen(open)
+		require.NoError(t, err)
+		require.True(t, authorizedI)
 	}
 
 	{
 		ci, di, fi, err := initiatorChannel.OpenTxs(state.OpenParams{Asset: asset, AssetLimit: assetLimit})
 		require.NoError(t, err)
 
-		ci, err = ci.AddSignatureDecorated(open.CloseSignatures...)
+		ci, err = ci.AddSignatureDecorated(initiatorChannel.OpenAgreement().CloseSignatures...)
```

Comment:
💭 I don't think this and the similar changes are necessary since `open` is in scope here and you just need to set it on line 67 above instead of the `_`. You don't have to change it if you prefer to do it this way, but just noticing this change seems unnecessary.
```suggestion
		ci, err = ci.AddSignatureDecorated(open.CloseSignatures...)
```

Reply:
I'd like to leave as using the `.OpenAgreement()` when submitting the open transaction to be really sure that the saved open agreement is correct (maybe down the road other code modifies the saved open agreement in an intermediate step). Even though the in scope `open` object should definitely be the same, just to be sure